### PR TITLE
util: add truncate package

### PR DESCRIPTION
**Pull request by creachadair**
Corresponding internal issue: https://github.com/tailscale/corp/issues/9642.

This package handles cases where we need to truncate human-readable text to fit a length constraint without leaving "ragged" multi-byte rune fragments at the end of the truncated value.

Change-Id: Id972135d1880485f41b1fedfb65c2b8cc012d416

### CODE REVIEW

**Review by soniaappasamy — APPROVED** [review]
Nice!

File: util/truncate/truncate.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package truncate provides a utility function for safely truncating UTF-8
+// strings to a fixed length, respecting multi-byte codepoints.
+package truncate
+
+// String returns a prefix of a UTF-8 string s, having length no greater than
```
[the remark this answers is not in the document]
Comment (creachadair) [reply util/truncate/truncate.go:8]: "Reports" is generally used for Boolean results. I don't feel super strongly about it, but I think in this context "returns" is the better word. (It's up to you, though)

[the remark this answers is not in the document]
Comment (creachadair) [reply util/truncate/truncate.go]: Rationale: One "reports" a status, one "returns" a result.

[the remark this answers is not in the document]
Comment (soniaappasamy) [reply util/truncate/truncate.go]: Fine with me!

**Pull request merged**